lib/storage/src/content_manager/toc.rs

```diff
@@ -745,7 +745,15 @@ impl TableOfContent {
         known_peers_set.insert(self.this_peer_id());
         let known_peers: Vec<_> = known_peers_set.into_iter().collect();
 
-        let shard_distribution = ShardDistributionProposal::new(shard_number, &known_peers, vec![]);
+        let mut current_distribution = Vec::new();
+        let known_collections = self.collections.read().await;
+        for (_, col) in known_collections.iter() {
+            let collection_distribution = col.shard_distribution(self.this_peer_id).await;
+            current_distribution.extend(collection_distribution);
+        }
+
+        let shard_distribution =
+            ShardDistributionProposal::new(shard_number, &known_peers, current_distribution);
```

Comment:
I think the version where collections are distributed independently was better for the following reason: collections are not guaranteed to be equal in size, so global balancing might create uneven distribution on points count view. In this case even distribution for individual collections would be more predictable